app/code/Magento/Customer/Model/Options.php

```diff
@@ -53,23 +57,43 @@ public function getNamePrefixOptions($store = null)
      */
     public function getNameSuffixOptions($store = null)
     {
-        return $this->_prepareNamePrefixSuffixOptions($this->addressHelper->getConfig('suffix_options', $store));
+        return $this->prepareNamePrefixSuffixOptions(
+            $this->addressHelper->getConfig('suffix_options', $store),
+            $this->addressHelper->getConfig('suffix_show', $store) == NooptreqSource::VALUE_OPTIONAL
```

Comment:
Unfortunately we will need to change this since we cannot add the new constants.